File: medeia-validator-java-examples/src/main/java/com/worldturner/medeia/examples/java/domain/Fixtures.java
package com.worldturner.medeia.examples.java.domain;

public class Fixtures {
    public static Person createValidPerson() {
        Person person = new Person();
        person.setFirstName("Yon");
        person.setLastName("Yonson");
        Address address = new Address();
        address.setRegion("Wisconsin");
        address.setStreet1("1 Lumberjack Road");
        address.setCity("Townsville");
        address.setPostalCode("34216");
        address.setCountry("USA");
        person.setAddress(address);
        return person;
    }

    public static Person createInvalidPerson() {
        Person person = createValidPerson();
        person.getAddress().setStreet1(null);
        return person;
    }
}

File: medeia-validator-kotlin-examples/src/main/kotlin/com/worldturner/medeia/examples/kotlin/domain/fixtures.kt
package com.worldturner.medeia.examples.kotlin.domain

fun createValidPersonFixture(): Person =
    Person(
        firstName = "Yon",
        lastName = "Yonson",
        address = Address(
            region = "Wisconsin",
            street1 = "1 Lumberjack Road",
            street2 = null,
            city = "Townsville",
            postalCode = "34216",
            country = "USA"
        )
    )

fun createInvalidPersonFixture(): Person {
    val person = createValidPersonFixture()
    return person.copy(address = person.address.copy(city = ""))
}